sonic_y_cable/y_cable.py

```diff
@@ -176,6 +177,67 @@ def check_read_side(physical_port):
     else:
         helper_logger.log_error(
             "Error: unknown status for checking which side regval = {} ".format(result))
+
+    return -1
+
+
+def check_mux_direction(physical_port):
+    """
+    This API specifically checks which side of the Y cable mux is currently point to
+    and returns either TOR A or TOR B. API basically reads 1 byte at upper page 4 offset 132
+    and checks which side the mux is pointing to
+
+
+    Register Specification of upper page 0x4 at offset 133 is documented below
+
+    Byte offset   bits     Name                           Description
+    133           7-0      MUX Switch Status Register     0x00 : MUX pointing at TOR#2, 0x01 MUX pointing at TOR#1 regardless of connection status
+
+    Args:
+         physical_port:
+             an Integer, the actual physical port connected to a Y cable
+
+    Returns:
+        an Integer, 1 if the mux is pointing to TOR A .
+                  , 2 if the mux is pointing to TOR B.
+                  , -1 if checking which side mux is pointing to API fails.
+    """
+
+    curr_offset = Y_CABLE_MUX_DIRECTION
+
+    if platform_chassis is not None:
+        result = platform_chassis.get_sfp(
+            physical_port).read_eeprom(curr_offset, 1)
```

Comment:
can this API block?

Reply:
No this has a timeout, so its non blocking